test/playlist-controller.test.js

```diff
@@ -2273,8 +2273,8 @@ QUnit.test('does not get stuck in a loop due to inconsistent network/caching', f
     assert.notOk(media1ResolvedPlaylist.excludeUntil, 'media1 not excluded');
     assert.equal(
       segmentRequest.uri.substring(segmentRequest.uri.length - 4),
-      '0.ts',
-      'requested first segment'
+      '1.ts',
```

Comment:
This is just a question about the previous implementation before this fix.. But should this have been a sign that we could have some repeated segments? Should we have always expected the second segment to be requested here?

Reply:
We should not request any of these segments in this specific test case scenario since they are already appended.  
This test does not reflect a real-world scenario, so I simply updated the numbers.

Reply:
I can add this check to ensure that we do not load the next segment if it is already appended, but it is an artificial scenario and an unexpected case.


```js
    if (nextMediaSequenceSyncInfo.isAppended) {
      // unexpected case
      return null;
    }
```